### PR TITLE
[SH1945][SH4623] - Creation of new login flow and adapting graphql 

### PR DESCRIPTION
**Pull request by 92Sam**
https://app.shortcut.com/masterventures/story/4623/as-a-backend-engineer-i-need-to-remove-xdv-from-use
 https://app.shortcut.com/masterventures/story/1945/as-a-frontend-developer-i-need-to-integrate-magic-link-into-the-sign-up-process-in-the-dashboard

[SH4623] 
[SH1945]

### CODE REVIEW

File: components/account-modal/LoginModal.tsx [hunk 1]
```diff
@@ -82,9 +82,13 @@ const LoginModal: FC<AccountModalProps> = ({ open, onLogin }: AccountModalProps)
...
 
       const res2 = await dispatch(doSetLoginMagicLink(didToken, values.email));
-      console.log(res2)
+      console.log("Respuesta 2 Obtener toquen", res2)
 
-      console.log(didToken);
+      if (res2) {
+        console.log("Token COmponente ----->", didToken);
+        const graphqlResponse = await dispatch(getProfileFromBearer());
+        console.log("graphqlResponse ", graphqlResponse)
```
Comment (CoolHandLuke88) [inline components/account-modal/LoginModal.tsx]: remove console.logs

File: components/account-modal/AccountModal.tsx [hunk 2]
```diff
@@ -20,17 +20,21 @@ const AccountModal: FC<AccountModalProps> = ({
   open,
 }: AccountModalProps) => {
   const dispatch = useDispatch();
-  const profileState: ProfileStateModel = useSelector(
-    (state: any) => state.profileReducer,
-  );
+  // const profileState: ProfileStateModel = useSelector(
+  //   (state: any) => state.profileReducer,
+  // );
```
Comment (CoolHandLuke88) [inline components/account-modal/AccountModal.tsx:25]: remove dead code

File: components/account-modal/AccountModal.tsx [hunk 3]
```diff
@@ -20,17 +20,21 @@ const AccountModal: FC<AccountModalProps> = ({
...
+  const profileReducer = useSelector(
+    (state: any) => state.profileReducer,
+  );
+
   useEffect(() => {
-    const bootstrapAsync = async () => {
+    /*const bootstrapAsync = async () => {
       if (profile.passphrase && profile.passphrase !== '') {
         const created = formatDateProfile(new Date());
         const accountName = (`${profile.name.toLocaleLowerCase()}${profile.lastName.toLocaleLowerCase()}`).replace(/(\s)/gi, '');
```
Comment (CoolHandLuke88) [inline components/account-modal/AccountModal.tsx]: remove dead code

File: components/account-modal/FormAccount.tsx [hunk 4]
```diff
@@ -131,7 +131,7 @@ const FormAccount: FC<FormAccountProps> = ({
             errors={errors}
           />
         )}
-      />
+      /> */}
```
Comment (CoolHandLuke88) [inline components/account-modal/FormAccount.tsx:134]: dead code

File: models/profileModel.ts [hunk 5]
```diff
@@ -1,3 +1,5 @@
+// import { User } from '../lib/dtos'
```
Comment (CoolHandLuke88) [inline models/profileModel.ts:1]: dead code

File: pages/_app.jsx [hunk 6]
```diff
@@ -70,18 +72,33 @@ function MyApp({ Component, pageProps }) {
...
+      //   account &&
+      //   !walletReducer.isDisconnecting
+      // ) {
+      //   //const getCurrentWallet = global.localStorage.getItem(account)
+      //   // if (getCurrentWallet) {
+      //   //   setOpenPassphraseModal(true)
+      //   // } else {
+      //   //   setOpenAccountModal(true)
+      //   // }countModal(true)
+      //   }
```
Comment (CoolHandLuke88) [inline pages/_app.jsx:124]: dead code

File: pages/_app.jsx [hunk 7]
```diff
@@ -102,11 +119,11 @@ function MyApp({ Component, pageProps }) {
           </PrivateLayout>
         )}
         <AccountModal open={openAccountModal} />
-        <PassphraseModal
+        {/* <PassphraseModal
           open={openPassphraseModal}
           errorPassphrase={errorPassphrase}
           setPassphrase={setPassphrase}
-        />
+        /> */}
```
Comment (CoolHandLuke88) [inline pages/_app.jsx:149]: dead code

File: components/account-modal/AccountModal.tsx [hunk 8]
```diff
@@ -34,59 +34,33 @@ const AccountModal: FC<AccountModalProps> = ({
...
   useEffect(() => {
-    /*const bootstrapAsync = async () => {
-      if (profile.passphrase && profile.passphrase !== '') {
-        const created = formatDateProfile(new Date());
-        const accountName = (`${profile.name.toLocaleLowerCase()}${profile.lastName.toLocaleLowerCase()}`).replace(/(\s)/gi, '');
-        const xdvWallet = new Wallet({ isWeb: true });
+    console.log("State of account On Account Modal - ", account)
+    console.log("State of Profile On Account Modal - ", profile)
+    console.log("State of Profile Reducer Global On Account Modal - ", profileReducer)
+    console.log(profile, "  -- -- -- -- --", profileReducer.did)
```
Comment (CoolHandLuke88) [inline components/account-modal/AccountModal.tsx:40]: remove console.logs

File: pages/_app.jsx [hunk 9]
```diff
@@ -3,12 +3,16 @@ import { useRouter } from 'next/router'
 import { Provider, useSelector, useDispatch } from 'react-redux'
 import { WalletProvider, useWallet } from 'react-binance-wallet'
 
-import PassphraseModal from '@/components/profile/PassphraseModal'
+// import PassphraseModal from '@/components/profile/PassphraseModal'
 import AccountModal from '@/components/account-modal/AccountModal'
 import PrivateLayout from '@/components/Layout/PrivateLayout'
-import getDidXdv from '../utils/xdv-universal-wallet'
+// import getDidXdv from '../utils/xdv-universal-wallet'
```
Comment (CoolHandLuke88) [inline pages/_app.jsx:9]: dead code

File: pages/_app.jsx [hunk 10]
```diff
@@ -33,33 +37,33 @@ function MyApp({ Component, pageProps }) {
...
+    //           setErrorPassphrase(true)
+    //         }
+    //       }
+    //     }
+    //   }
 
-      bootstrapAsync()
-    }, [passphrase])
+    //   bootstrapAsync()
+    // }, [passphrase])
```
Comment (CoolHandLuke88) [inline pages/_app.jsx:66]: dead code

File: pages/profile.tsx [hunk 11]
```diff
@@ -25,41 +25,41 @@ const Profile: FC = () => {
...
+      //   ...profile,
+      //   ...values,
+      //   did: provider.did,
+      //   created: profileData.created,
+      // };
 
-      dispatch(doSetProfile(currentProfile));
-      setProfile(currentProfile);
+      // dispatch(doSetProfile(currentProfile));
+      // setProfile(currentProfile);
```
Comment (CoolHandLuke88) [inline pages/profile.tsx:62]: dead code

**Review by CoolHandLuke88 — APPROVED** [review]
+1 address/ignore comments, but we should remove console.logs if not needed and dead code if no longer used.

**Comment by 92Sam** [comment]
@CoolHandLuke88 could you check again, please. 

And thanks a lot.

File: pages/_app.jsx [hunk 12]
```diff
@@ -4,9 +4,8 @@ import { Provider, useSelector, useDispatch } from 'react-redux'
 import { WalletProvider, useWallet } from 'react-binance-wallet'
 
 // import PassphraseModal from '@/components/profile/PassphraseModal'
```
Comment (CoolHandLuke88) [inline pages/_app.jsx]: dead code

File: pages/profile.tsx [hunk 13]
```diff
@@ -25,39 +24,6 @@ const Profile: FC = () => {
...
-      // global.localStorage.setItem(currentWallet, JSON.stringify(walletStorage));
-      // const currentProfile = {
-      //   ...profile,
-      //   ...values,
-      //   did: provider.did,
-      //   created: profileData.created,
-      // };
-
       // dispatch(doSetProfile(currentProfile));
       // setProfile(currentProfile);
```
Comment (CoolHandLuke88) [inline pages/profile.tsx]: dead code

File: pages/new-agreement.tsx [hunk 14]
```diff
@@ -68,10 +68,14 @@ const NewAgreement: NextPage<NewAgreementProps> = ({ templateTypeCode }) => {
...
 
-  const {
-    name, email, did,
-  } = useSelector(
-    (state: { profileReducer: ProfileStateModel }) => state.profileReducer.profile,
+  // const {
+  //   name, email, did,
+  // } = useSelector(
+  //   (state: { profileReducer: ProfileStateModel }) => state.profileReducer.profile,
+  // );
```
Comment (CoolHandLuke88) [inline pages/new-agreement.tsx]: dead code

File: models/profileModel.ts [hunk 15]
```diff
@@ -2,17 +2,20 @@
...
+  firstName?: string
+  lastName?: string
+  streetAddress?: string
+  profileName?: string
+
 }
 
 interface WalletExtend {
-  profileName?: string
+  // profileName?: string
```
Comment (CoolHandLuke88) [inline models/profileModel.ts:18]: dead code

**Review by CoolHandLuke88 — APPROVED** [review]
+1 address/ignore, left more comments for `dead code`

File: components/account-modal/FormAccount.tsx [hunk 16]
```diff
@@ -2,24 +2,26 @@ import React, { FC, useRef } from 'react';
...
 import { ErrorMessage } from '@hookform/error-message';
 import classNames from 'classnames';
-import StackedPassword from '@/components/reusable/StackedPassword';
 import StackedInput from '../reusable/StackedInput';
 import ProfileModel from '../../models/profileModel';
 
 interface FormAccountProps {
   setProfile: any;
   profile: any;
+  user: any;
```
Comment (zubietaroberto) [inline components/account-modal/FormAccount.tsx:11]: Avoid using `any`

File: components/account-modal/FormLogin.tsx [hunk 17]
```diff
@@ -0,0 +1,56 @@
+import React, { FC } from 'react';
+import { useForm } from 'react-hook-form';
+import { ErrorMessage } from '@hookform/error-message';
+import classNames from 'classnames';
+import StackedInput from '../reusable/StackedInput';
+import ProfileModel from '../../models/profileModel';
+
+interface FormAccountProps {
+  login: any;
```
Comment (zubietaroberto) [inline components/account-modal/FormLogin.tsx:9]: Avoid using `any`

File: components/account-modal/LoginModal.tsx [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+import { PdModal, PdModalBody } from '@/pdComponents';
+
+import ProfileModel from '@/models/profileModel';
+import { setCurrentWallet } from '../../redux/actions/wallet';
+import { doSetProfile, doSetLoginMagicLink, getProfileFromBearer } from '../../redux/actions/profile';
+import helper from '../../utils/helper';
+import FormLogin from './FormLogin';
+interface AccountModalProps {
+  open: boolean;
+  onLogin: any;
```
Comment (zubietaroberto) [inline components/account-modal/LoginModal.tsx:16]: Avoid using `any`

File: models/profileModel.ts [hunk 19]
```diff
@@ -1,16 +1,21 @@
...
```
Comment (zubietaroberto) [inline models/profileModel.ts:18]: Dead code

File: package.json [hunk 20]
```diff
@@ -17,18 +17,21 @@
     "@binance-chain/bsc-use-wallet": "^0.8.1",
     "@connext/vector-modal": "^1.0.5-beta.0",
     "@hookform/error-message": "0.0.5",
+    "@magic-sdk/admin": "^1.3.0",
```
Comment (zubietaroberto) [inline package.json:20]: This library I think is only required in the server. Please verify.

File: redux/actions/profile.ts [hunk 21]
```diff
@@ -1,8 +1,77 @@
...
+      payload: userData,
+    })
+    return Promise.resolve(userData)
+  } catch (err) {
+    return Promise.reject(err)
+  }
 }
 
-export default doSetProfile
+// export default doSetProfile
```
Comment (zubietaroberto) [inline redux/actions/profile.ts:77]: Unused line

File: redux/actions/wallet.ts [hunk 22]
```diff
@@ -1,42 +1,86 @@
...
+  router: NextRouter,
+  query: any,
+) => async (dispatch: any) => {
+  // eslint-disable-next-line no-undef
+  try {
+    const walletGraphqlValues: WalletAddressInput = {
+      address: currentWallet,
+      network: NetworkName.RINKEBY,
+    }
+    console.log('Wallet Values ------|', walletGraphqlValues)
```
Comment (zubietaroberto) [inline redux/actions/wallet.ts:27]: Remove console.log

File: redux/actions/wallet.ts [hunk 23]
```diff
@@ -1,42 +1,86 @@
...
+    const userWallet = await initializeApollo()
+      .mutate({
+        mutation: MUTATION_WALLET,
+        variables: { input: { ...walletGraphqlValues } },
+      })
+      .then((response) => {
+        return response
+      })
+
+    console.log('User Wallet ------|', userWallet)
```
Comment (zubietaroberto) [inline redux/actions/wallet.ts:37]: Remove console.log

File: sass/module/pages/_index.scss [hunk 24]
```diff
@@ -65,3 +65,32 @@
...
+      margin-top: 15px;
+      margin-bottom: 40px;
+      color: $blue;
+      font-family: Circular Std;
+      font-size: 18px !important;
+      line-height: 22.77px;
+      text-align: center !important;
+    }
+  }
+}
```
Comment (zubietaroberto) [inline sass/module/pages/_index.scss:96]: Instead of adding it here, please add this as a SCSS module, like the other components: `LoginModal.module.scss`

**Review by zubietaroberto — CHANGES_REQUESTED** [review]
Missing some changes

File: package.json (hunk 20 above)
Comment (92Sam) [reply package.json:20]: dropped line

File: redux/actions/wallet.ts (hunk 23 above)
Comment (92Sam) [reply redux/actions/wallet.ts:37]: dropped

File: sass/module/pages/_index.scss (hunk 24 above)
Comment (92Sam) [reply sass/module/pages/_index.scss:96]: done.

**Comment by bstahlhood** [comment]
@92Sam Make sure you rebase against `main`

**Comment by 92Sam** [comment]
@bstahlhood rebasing the new changes incoming.

**Review by CoolHandLuke88 — APPROVED** [review]

**Review by zubietaroberto — APPROVED** [review]
Approved changes

**Pull request merged**